src/HostRouter.ts

```diff
@@ -41,7 +47,7 @@ export class HostRouter {
       }
     });
 
-    return clientTarget;
+    return clientTarget.id ? clientTarget : null;
```

Comment:
This is a weird way to do this. I would just start with `clientTarget` initialized to null and leave the return statement as is.